numba/np/arrayobj.py

```diff
@@ -5042,8 +5042,12 @@ def impl(ary, indices_or_sections, axis=0):
                 raise ValueError(
                     "array split does not result in an equal division"
                 )
+            indices = np.cumsum(np.array(
+                [l + 1] * rem +
+                [l] * (indices_or_sections - rem - 1)
+            ))
             return np.array_split(
-                ary, np.arange(l, ary.shape[axis], l), axis=axis
+                ary, indices, axis=axis
```

Comment:
This is the same code as in `np.array_split` when an integer is passed for indices or sections. If I were to pass the `indices_or_sections` to `np.array_split` instead, would the compiler know it could reuse the results of the divmod call on 5040? This is with the assumption everything gets inlined.

Reply:
Quickly looking at the assembly suggests it's struggling to inline `array_split` and so doesn't spot the `idiv` on the same vars.

Reply:
Looking at this some more, I think passing through `indices_or_sections` is probably a good thing on two fronts, 1. cleans the code up a bit, 2. eventually the compiler might inline this to avoid the `divmod` but more importantly, passing it through avoids creating heap objects like arrays and lists when forming up `indices`. Are you happy to just pass it?

Reply:
Sure! Done now.